### PR TITLE
output: Support record_accessor in buffer's chunk keys

### PR DESCRIPTION
**Pull request by repeatedly**
This patch add record_accessor's dot-notation support in chunk keys.
To support bracket-notation, we need to consider relax validation in many places.
So it is the future work.

### CODE REVIEW

**Comment by repeatedly** [comment]
@mururu Could you check this?

File: lib/fluent/plugin/output.rb [hunk 1]
```diff
@@ -36,7 +37,7 @@ class Output < Base
       helpers_internal :thread, :retry_state
 
       CHUNK_KEY_PATTERN = /^[-_.@a-zA-Z0-9]+$/
-      CHUNK_KEY_PLACEHOLDER_PATTERN = /\$\{[-_.@a-zA-Z0-9]+\}/
+      CHUNK_KEY_PLACEHOLDER_PATTERN = /\$\{[-_.@$a-zA-Z0-9]+\}/
```
Comment (mururu) [inline lib/fluent/plugin/output.rb]: Why do we have to add `$`?

Comment (repeatedly) [inline lib/fluent/plugin/output.rb]: Without `$`, we can't put `$` inside placeholders like `${$.nested.key}`.

Comment (mururu) [inline lib/fluent/plugin/output.rb]: I got it.

**Pull request merged**